src/Rules/DateBetween.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Spatie\ValidationRules\Rules;
+
+use Carbon\Carbon;
+use Illuminate\Contracts\Validation\Rule;
+
+class DateBetween implements Rule
+{
+    /** @var \Carbon\Carbon */
+    protected $start;
+
+    /** @var \Carbon\Carbon */
+    protected $end;
+
+    /** @var bool */
+    protected $orEquals = false;
+
+    public function __construct(Carbon $start, Carbon $end)
+    {
+        $this->start = $start->copy();
+
+        $this->end = $end->copy();
+    }
+
+    public function orEquals(bool $orEquals = true): DateBetween
```

Comment:
Let's not pass booleans, but make two dedictated methods for this. `includingBoundaries` and `excludingBoundaries` or something better.